lib/web_ui/lib/src/engine/canvaskit/canvaskit_api.dart

```diff
@@ -1859,17 +1859,20 @@ class SkLineMetrics {
 
 @JS()
 @anonymous
-class SkParagraph {
+@staticInterop
+class SkParagraph {}
+
+extension SkParagraphExtension on SkParagraph {
   external double getAlphabeticBaseline();
   external bool didExceedMaxLines();
   external double getHeight();
   external double getIdeographicBaseline();
-  external List<SkLineMetrics> getLineMetrics();
+  external /* List<SkLineMetrics> */ dynamic getLineMetrics();
```

Comment:
Could you explain this change? This is a limitation of @staticInterop somehow?

Reply:
This issue is unfortunate. As far as I understand things, the older interop model wasn't type checking, so this was effectively dynamic before. Static interop does type checking though, so now this seems to fail.

I *think* these methods must be returning true JS arrays, hence they are typed as JSArray<dynamic>, as opposed to Dart Lists.

I'm going to try and figure out with @srujzs a way to possibly make this work because it is certainly not ideal.

Reply:
Though, it's worth pointing  in the case of objects originating in JS and flowing to Dart this may not be solvable in a performant way. We might be able to return a List<dynamic> or possibly List<Object> though. Will dig in a bit more and discuss with others and report back.

Reply:
It's worth noting that the old type was not truthful, objects incoming from JavaScript are not reified with Dart type parameters, so they cannot contain type parameters altogether. As @joshualitt noted, the old js-interop never checked this and allowed invalid types to leak through the system.

I'd expect `external List getLineMetrics()` to work though. @joshualitt - does that give you any errors?

For many that would be sufficient as a safe API. The alternative is to use a list cast wrapper:

```dart
  List<SkLineMetrics> getLineMetrics() =>
      callMethod<List>(this, 'getLineMetrics', [])
        .cast<SkLineMetrics>();
```

but most users will prefer the low-cost of not wrapping the list.

Reply:
Okay, I spoke with @yjbanov and he expressed a preference for returning `List<Object?>` and then the caller can choose how to deal with the issue. We discussed returning `List<dynamic>` but because Flutter forbids implicit casts from dynamic, this seemed like not an ideal option.

In the context of this CL, we return a `List<Object?>` and then cast on demand as the items in the list are needed. This is a tiny bit more verbose, but saves us the minor cost of the wrapper object.

I'm going to land this now, but it would be easy to revisit this decision if people feel we should go a different direction, i.e. using the cast wrapper, or dynamic, or `trustTypes` mode.